helen/action-wordpress-plugin-deploy
# Replace `--exclude-from` with `--filter` to allow for `.gitignore` syntax in `.distignore`

### Description of the Change
Swap out the `--exclude-from` flag with the `--filter` flag to allow for the full .gitignore syntax to be used in the .distignore file. This change is to provide compatibility for breaking changes in the WP-CLI `dist-archive` command.

https://github.com/wp-cli/dist-archive-command/pull/78

I think this change might require a release by the dist-archive repository before being needed. At the time of opening this pull request, the last release was 2020 yet more items have been merged into [main branch](https://github.com/wp-cli/dist-archive-command/tree/main).

Closes #139 

### How to test the Change
1. Create a `.distignore` to the plugin root
```
touch .distignore
```
2. Create a text file named `important.txt`
```
touch important.txt
```
4. Add a rule to the `.distignore` to ignore all `.txt` files but to include `important.txt`
```
echo -e '*.txt\n!important.txt' > .distignore
```
5. Commit and deploy
6. Verify `important.txt` is at the destination

### Changelog Entry
> Changed - replace --exclude-from with --filter when using .distignore to allow for full .gitignore syntax


### Credits
Props @claytoncollie 


### Checklist:


- [] I have added tests to cover my change.
